qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -575,6 +583,12 @@ def add_schedule(
         if not isinstance(schedule, ScheduleBlock):
             raise CalibrationError(f"{schedule.name} is not a ScheduleBlock.")
 
+        if len(schedule.references) != len(schedule.references.unassigned()):
```

Comment:
I didn't notice anything in the added code that relied on all the references being unassigned. Couldn't you just deal only with the unassigned references in `get_schedule` and leave the assigned ones alone?

Reply:
This could get complicated with parameter name scoping but I have not looked too deeply into that. In general, I'm very hesitant to allow calibrations too much flexibility since the class is already complex enough. I suggest leaving as is and reconsidering when we reexamine the internals of `Calibrations`.

Reply:
This could get complicated with parameter name scoping but I have not looked too deeply into that. In general, I'm very hesitant to allow calibrations too much flexibility since the class is already complex enough. I suggest leaving as is and reconsidering when we reexamine the internals of `Calibrations`.